exonum-java-binding/core/src/test/java/com/exonum/binding/runtime/Pf4jServiceLoaderIntegrationTest.java

```diff
@@ -16,74 +16,78 @@
 
 package com.exonum.binding.runtime;
 
+import static java.util.Arrays.asList;
 import static java.util.Collections.emptyList;
-import static java.util.stream.Collectors.toList;
+import static java.util.Collections.singletonList;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
 
-import com.exonum.binding.service.AbstractServiceModule;
 import com.exonum.binding.service.ServiceModule;
+import com.exonum.binding.test.runtime.ServiceArtifactBuilder;
 import com.google.common.collect.ImmutableList;
+import java.io.IOException;
 import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
 import java.util.Optional;
 import java.util.function.Supplier;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
 import org.junit.jupiter.params.provider.ValueSource;
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
+import org.pf4j.DefaultPluginManager;
+import org.pf4j.Plugin;
 import org.pf4j.PluginManager;
-import org.pf4j.PluginState;
 
-@SuppressWarnings("WeakerAccess")
 @ExtendWith(MockitoExtension.class)
-class Pf4jServiceLoaderTest {
+class Pf4jServiceLoaderIntegrationTest {
 
-  private static final Path ARTIFACT_LOCATION = Paths.get("/tmp/service.jar");
   private static final String PLUGIN_ID = "com.acme:foo-service:1.0.1";
 
-  @Mock
   private PluginManager pluginManager;
 
-  @InjectMocks
   private Pf4jServiceLoader serviceLoader;
+  private Path artifactLocation;
+
+  @BeforeEach
+  void setUp(@TempDir Path tmp) {
+    pluginManager = spy(new DefaultPluginManager());
```

Comment:
Can be replaced by `@Spy`.
Also, as far as I remember `MockitoExtension` is used for annotations processing and it can be removed such as there are no annotations left.

Reply:
> for annotations processing

Also to detect unnecessary stubbings (but such don't seem possible with `spy`).

Reply:
yeap. So I propose to use `@Spy` or to remove the extension.